File: presets/MidiKit/Lua/Copy Ch1 CC to Ch2.lua
--[[
@target stoermelder MIDI-KIT
@engine minilua@v1
@author stoermelder
@description Duplicates all CC messages on channel 1 on channel 2
--]]

midi.onMessage = function(midiPort, msg)
    if midi.getChannel(msg) == 1 then
        if midi.isCc(msg) then
            local msg2 = midi.create()
            midi.setCc(msg2, 2, midi.getControl(msg), midi.getValue(msg))
            midiOut.send(msg2)
        end
    end
    midiOut.send(msg)
end

File: presets/MidiKit/JavaScript/Copy Ch1 CC to Ch2.js
/**
 * @target stoermelder MIDI-KIT
 * @engine Elk
 * @author stoermelder
 * @description Duplicates all CC messages on channel 1 on channel 2
 */

onMidiMessage = function(midiPort, msg) {
    if (midi.getChannel(msg) === 1) {
        if (midi.isCc(msg)) {
            let msg2 = midi.create();
            midi.setCc(msg2, 2, midi.getNote(msg), midi.getValue(msg));
            midiOut.send(msg2);
        }
    }
    midiOut.send(msg);
};